### ISSUE TITLE
Don't mutate the options object

### ISSUE THREAD

**Issue by TehShrike** [TehShrike/deepmerge#166]
[right now](https://github.com/TehShrike/deepmerge/blob/ad8cfddaa8f4b79f5e2c6943445a8b42377ec495/index.js#L58-L59) if you pass in an options object, and you don't specify any of the optional properties, the code just assigns those properties on the object you passed in.  This is rude and just poor form in general.

The easiest way that comes to mind to fix this would be to wait until #161 and then use the object spread syntax to combine the input options and a default options object into a new object.

**Comment by macdja38** [comment]
```
	options = Object.assign({}, { arrayMerge: defaultArrayMerge , isMergeableObject: defaultIsMergeableObject }, options)
```
Would do pretty much the same thing but without requiring es6, at the cost of a few more characters.

**Comment by TehShrike** [comment]
Yeah, that would work.  If anyone wants to do that in the near future we can put it out as a feature release before 5.0.0

**Comment by TehShrike** [comment]
Published as 4.1.1

**Comment by TehShrike** [comment]
Whoops, switching to Object.assign breaks support for IE11.  I'll move this change over to the v5 branch.
